ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/CommandTransactionChecks.scala

```diff
@@ -423,6 +426,98 @@ abstract class CommandTransactionChecks
         }
       }
 
+      "permit fetching a divulged contract" in forAllMatchingFixtures {
+        case TestFixture(SandboxInMemory, ctx) =>
+        def pf(label: String, party: String) =
+          RecordField(label, Some(Value(Value.Sum.Party(party))))
+        val odArgs = Seq(pf("owner", owner), pf("delegate", delegate))
+        val delegatedCreate = simpleCreate(
+          ctx,
+          cid("SDVl3"),
+          owner,
+          templateIds.delegated,
+          Record(Some(templateIds.delegated), Seq(pf("owner", owner))))
+        val delegationCreate = simpleCreate(
+          ctx,
+          cid("SDVl4"),
+          owner,
+          templateIds.delegation,
+          Record(Some(templateIds.delegation), odArgs))
+        val showIdCreate = simpleCreate(
+          ctx,
+          cid("SDVl5"),
+          owner,
+          templateIds.showDelegated,
+          Record(Some(templateIds.showDelegated), odArgs))
+        val exerciseOfFetch = for {
+          delegatedEv <- delegatedCreate
+          delegationEv <- delegationCreate
+          showIdEv <- showIdCreate
+          fetchArg = Record(
+            None,
+            Seq(RecordField("", Some(Value(Value.Sum.ContractId(delegatedEv.contractId))))))
+          showResult <- failingExercise(
+            ctx,
+            cid("SDVl6"),
+            submitter = owner,
+            template = templateIds.showDelegated,
+            contractId = showIdEv.contractId,
+            choice = "ShowIt",
+            arg = Value(Value.Sum.Record(fetchArg)),
+            Code.OK,
+            pattern = ""
+          )
+          fetchResult <- failingExercise(
```

Comment:
Maybe a comment here saying the `delegate` cannot fetch Delegation contracts, because it is not an owner.
But yeah, I don't see that we are testing anything that `delegate` can do in this test.

Reply:
`delegate` _can_ fetch because it has been divulged to it in the `ShowIt` exercise that owner did in the previous transaction.